packages/flutter/lib/src/material/checkbox_theme.dart

```diff
@@ -0,0 +1,234 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:ui' show lerpDouble;
+
+import 'package:flutter/foundation.dart';
+import 'package:flutter/rendering.dart';
+import 'package:flutter/widgets.dart';
+
+import 'material_state.dart';
+import 'theme.dart';
+import 'theme_data.dart';
+
+/// Defines default property values for descendant [Checkbox] widgets.
+///
+/// Descendant widgets obtain the current [CheckboxThemeData] object using
+/// `CheckboxTheme.of(context)`. Instances of [CheckboxThemeData] can be
+/// customized with [CheckboxThemeData.copyWith].
+///
+/// Typically a [CheckboxThemeData] is specified as part of the overall [Theme]
+/// with [ThemeData.checkboxTheme].
+///
+/// All [CheckboxThemeData] properties are `null` by default. When null, the
+/// [Checkbox] will use the values from [ThemeData] if they exist, otherwise it
+/// will provide its own defaults based on the overall [Theme]'s colorScheme.
+/// See the individual [Checkbox] properties for details.
+///
+/// See also:
+///
+///  * [ThemeData], which describes the overall theme information for the
+///    application.
+@immutable
+class CheckboxThemeData with Diagnosticable {
+  /// Creates a theme that can be used for [ThemeData.checkboxTheme].
+  const CheckboxThemeData({
+    this.mouseCursor,
+    this.fillColor,
+    this.checkColor,
+    this.overlayColor,
+    this.splashRadius,
+    this.materialTapTargetSize,
+    this.visualDensity,
+  });
+
+  /// {@macro flutter.material.checkbox.mouseCursor}
+  ///
+  /// If specified, overrides the default value of [Checkbox.mouseCursor].
+  final MaterialStateProperty<MouseCursor?>? mouseCursor;
+
+  /// {@macro flutter.material.checkbox.fillColor}
+  ///
+  /// If specified, overrides the default value of [Checkbox.fillColor].
+  final MaterialStateProperty<Color?>? fillColor;
+
+  /// {@macro flutter.material.checkbox.checkColor}
+  ///
+  /// If specified, overrides the default value of [Checkbox.checkColor].
+  final Color? checkColor;
```

Comment:
Should this be a `MaterialState<Color?>?` ? Apps might want to specify the disabled and selected check color and (maybe) the unselected check color should default to transparent. We wouldn't draw it at all in that case, but it means that one could also configure the check to appear when the checkbox is not selected?

Reply:
I can add the functionality to make it a material state aware color.

To configure if the check is able to appear or not, seems like something that could be done in a separate feature request issue/PR. I can create a issue for this.

Reply:
Added this now.